tests/Aspire.Hosting.Qdrant.Tests/QdrantPublicApiTests.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Xunit;
+
+namespace Aspire.Hosting.Qdrant.Tests;
+
+public class QdrantPublicApiTests
+{
+    #region QdrantBuilderExtensions
+
+    [Fact]
+    public void AddQdrantContainerShouldThrowsWhenBuilderIsNull()
+    {
+        IDistributedApplicationBuilder builder = null!;
+        const string name = "Qdrant";
+
+        var action = () => builder.AddQdrant(name);
+
+        Assert.Multiple(() =>
+        {
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal(nameof(builder), exception.ParamName);
+        });
+    }
+
+    [Fact]
+    public void AddQdrantContainerShouldThrowsWhenNameIsNull()
+    {
+        var builder = DistributedApplication.CreateBuilder([]);
+        string name = null!;
+
+        var action = () => builder.AddQdrant(name);
+
+        Assert.Multiple(() =>
+        {
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal(nameof(name), exception.ParamName);
+        });
+    }
+
+    [Fact]
+    public void WithDataVolumeShouldThrowsWhenBuilderIsNull()
+    {
+        IResourceBuilder<QdrantServerResource> builder = null!;
+
+        var action = () => builder.WithDataVolume();
+
+        Assert.Multiple(() =>
+        {
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal(nameof(builder), exception.ParamName);
+        });
+    }
+
+    [Fact]
+    public void WithDataBindMountShouldThrowsWhenBuilderIsNull()
+    {
+        IResourceBuilder<QdrantServerResource> builder = null!;
+        const string source = "/qdrant/storage";
+
+        var action = () => builder.WithDataBindMount(source);
+
+        Assert.Multiple(() =>
+        {
+            var exception = Assert.Throws<ArgumentNullException>(action);
+            Assert.Equal(nameof(builder), exception.ParamName);
+        });
+    }
+
+    [Fact]
+    public void WithDataBindMountShouldThrowsWhenNameIsNull()
+    {
+        var distributedApplicationBuilder = DistributedApplication.CreateBuilder([]);
+        const string name = "Qdrant";
+        var apiKeyParameter = ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(distributedApplicationBuilder, $"{name}-Key", special: false);
+        var resource = new QdrantServerResource(name, apiKeyParameter);
+        var builder = distributedApplicationBuilder.AddResource(resource);
+        string source = null!;
+
+        var action = () => builder.WithDataBindMount(source);
```

Comment:
Do we need all this or would something like this be enough?:
```csharp
        IResourceBuilder<QdrantServerResource> builder = null!;
        string source = null!;
        var action = () => builder.WithDataBindMount(source);
```

@eerhardt @sebastienros thoughts?

Reply:
I agree, it doesn't need to be that complicated. Even the `Multiple` usage in all these tests, since the second assertion depends on the first, we don't care about the second result if the first one failed.

Reply:
> Multiple

Indeed, I will remove Multiple.

Reply:
> Do we need all this or would something like this be enough?:
> 
> ```cs
>         IResourceBuilder<QdrantServerResource> builder = null!;
>         string source = null!;
>         var action = () => builder.WithDataBindMount(source);
> ```
> 
> @eerhardt @sebastienros thoughts?

But in this case. We will not handle the error when builder 'is not null' and source 'is null'

Reply:
Full test

```c#
var builder = TestDistributedApplicationBuilder.Create();
var qdrant = builder.AddQdrant("qdrant");
string source = null!;

var action = () => qdrant.WithDataBindMount(source);
var exception = Assert.Throws<ArgumentNullException>(action);
Assert.Equal(nameof(source), exception.ParamName);
```